File: 3329-find-the-length-of-the-longest-common-prefix/find-the-length-of-the-longest-common-prefix.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        int sz1 = arr1.size(), sz2 = arr2.size(), tmp, ans = 0;
        unordered_map<int, int> mp1, mp2;

        for(int i=0; i<sz1; i++){
            tmp = arr1[i];
            while(tmp){
                mp1[tmp]++;
                tmp /= 10;
            }
        }

        for(int i=0; i<sz2; i++){
            tmp = arr2[i];
            while(tmp){
                mp2[tmp]++;
                tmp /= 10;
            }
        }

        for(auto i : mp1){
            if(mp2.find(i.first) != mp2.end()){
                ans = max(ans, i.first);
            }
        }

        int cnt = 0;
        while(ans){
            cnt++;
            ans/=10;
        }

        return cnt;
    }
};

File: 3329-find-the-length-of-the-longest-common-prefix/find-the-length-of-the-longest-common-prefix.java
class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {
        int sz1 = arr1.length, sz2 = arr2.length, maxi = Integer.MIN_VALUE, ans = 0;
        Set<Integer> st = new HashSet<>();

        for(int i=0; i<sz1; i++) {
            int n = arr1[i];
            while(n > 0) {
                st.add(n);
                n /= 10;
            }
        }

        for(int i=0; i<sz2; i++){
            int n = arr2[i];
            while(n > 0) {
                if(st.contains(n)) {
                    maxi = maxi > n ? maxi : n;
                }
                n /= 10;
            }
        }

        while(maxi > 0) {
            ans++;
            maxi /= 10;
        }

        return ans;
    }
}